### PR TITLE
Fix Issue #1386

### PR DESCRIPTION
**Pull request by jackyq2015**
parse the time end with Z

### CODE REVIEW

File: main/src/com/google/refine/util/ParsingUtilities.java [hunk 1]
```diff
@@ -193,6 +195,10 @@ static public OffsetDateTime stringToDate(String s) {
     }
     
     static public LocalDateTime stringToLocalDate(String s) {
-      return LocalDateTime.parse(s, DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+        if (s.endsWith("Z")) {          // UTC time
+            Instant instant = Instant.parse(s);
+            return LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+        } else
```
Comment (ostephens) [inline main/src/com/google/refine/util/ParsingUtilities.java]: Missing brace

**Comment by ostephens** [comment]
Once that syntax issue fixed then this PR solves the immediate problem. Just wondering - are we sure that the only dates out there in older projects are of the format yyyy-MM-ddThh:MM:ssZ ? Could they ever be expressed with other timezone data - e.g. the Java ISO_DATE_TIME format which allows datetimes like:

'2011-12-03T10:15:30+01:00[Europe/Paris]'

If we are sure it is only those ending in Z that are problematic then that's OK, just wondering if we need a more comprehensive solution.

In relation I think it would be good to catch and deal with situations where the string->datetime conversion fails silently (which is essentially what happened in this case) so that the project isn't hidden from view. Potentially have a check for a valid datetime before trying to create the ProjectMetdata object, and fallback to using current date if the one parsed from string is invalid/null/zero length?

**Comment by jackyq2015** [comment]
yes, I am sure that is the only format. there is a formatter defined for iso8601 in the parsing utility class.

File: main/tests/server/src/com/google/refine/tests/util/ParsingUtilitiesTests.java [hunk 2]
```diff
@@ -107,4 +110,25 @@ public void parseProjectBeforeJDK8() {
...
+    public void strSubstitutorTest() {
+        Map<String, String> data = new HashMap<String, String>(6);
+        
+        data.put("value", "1234");
+        data.put("field_format", "String");
+        
+        StrSubstitutor sub = new StrSubstitutor(data);
+        String message = "The value ${value} in row ${row_number} and column ${column_number} is not type ${field_type} and format ${field_format}";
+        String result = sub.replace(message);
+        System.out.println("XXXXXXXX:" + result);
```
Comment (wetneb) [inline main/tests/server/src/com/google/refine/tests/util/ParsingUtilitiesTests.java]: I wouldn't print anything like this in a unit test.

**Pull request merged**